src/Sentry/GlobalSessionManager.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Sentry.Extensibility;
+using Sentry.Internal;
+
+namespace Sentry
+{
+    // AKA client mode
+    internal class GlobalSessionManager : ISessionManager
+    {
+        private readonly object _lock = new();
+        private readonly SentryOptions _options;
+
+        private string? _cachedInstallationId;
+        private Session? _currentSession;
+
+        public GlobalSessionManager(SentryOptions options)
+        {
+            _options = options;
+        }
+
+        private string GetInstallationId()
+        {
+            // Prevent race conditions that could cause us to generate multiple IDs.
+            // Note: this has to be synchronized across multiple processes too.
+            lock (_lock)
+            {
+                // Avoid IO if possible
+                if (!string.IsNullOrWhiteSpace(_cachedInstallationId))
+                {
+                    return _cachedInstallationId;
+                }
+
+                var filePath = Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
```

Comment:
We can't really assume there'll be access to disk I'm afraid. We could have a logic such as:

Rely on `CacheDirectoryPath`. If it's null, lets try to get the app data directory (what you did here, though I'd be afraid we might need an opt out, and definitely deal with errors that might throw from attempting to do I/O here).

If that fails we could hash `Environment.MachineName` or something else else cheap.
Lastly we could finally fallback to something you doing before like the mac address from the first NIC.

Think this could run on Wasm, Mobile, Unity (IL2CPP) on xbox, ps4 or whatever else they do. We are not testing any of those platforms right now but we can take into account that the code must be as portable as possible.

Reply:
Makes sense. So how about:

1. Use cache directory
2. Use app data
3. Use MAC address

I don't think there's much difference between machine name and MAC address in terms of cheapness, we do it once per runtime anyway.

Reply:
Yeah I agree that's a good approach